### DISCUSSION TITLE
gitbook &cloudflare  error

### DISCUSSION THREAD

**Discussion by YixiaoOrg** [GitbookIO/community#52] [2. Published content] [answered]
I used the domain name read.yixiao.org hosted on cloudflare to resolve to gitbook, and opened the CDN, and then modified the resolution, but it did not resolve to gitbook, but no matter how I modified the resolution, it was still redirected to gitbooks.

similar https://community.cloudflare.com/t/subdomain-goes-to-gitbook-but-not-set/377558
Please spare my domain name   read.yixiao.org
thanks

**Comment by rodrcastro** [comment]
Hey @StuPYX. I've removed your domain from Cloudflare, and it should work as expected now.

**Comment by jpreynat** [comment, marked as answer]
@StuPYX Sorry for the trouble. I just wanted to give some additional context on this specific issue as it has been a recurring problem that we are aware of but sadly cannot be easily resolved.

TL;DR: To regain complete control of your custom hostname after it has been registered on app.gitbook.com, remove this custom hostname from our platform.

For the longer explanation:
We use Cloudflare to serve HTTPS traffic for all custom hostnames configured by our users.

When a user configures a custom hostname, they point their DNS via CNAME to one of our domains (which, at the end of the chain points to Cloudflare). We then request Cloudflare (using their Cloudflare for SaaS product) to generate an SSL certificate for this hostname and serve the traffic properly.

When users move away from GitBook, they often don't remove their content from GitBook and only change the DNS on their side. We don't request to remove the hostname from Cloudflare for SaaS until the content is deleted from GitBook, as the goal is to avoid breaking links for URLs that are still pointing to GitBook.

We'd expect Cloudflare to always use the DNS setup of the domain as the primary factor for deciding where to route the traffic.

We don't know the rationale behind why Cloudflare routing continues internally routing the traffic to GitBook when the domain is no longer pointing to the GitBook hostname. But it is not us doing that intentionally.
